### DISCUSSION TITLE
外链表Oracle日期字段类型不能作为查询条件（ODBC和JDBC）

### DISCUSSION THREAD

**Discussion by stevqin** [apache/doris#15069] [RFC]
外链表Oracle日期字段类型不能作为查询条件；
ODBC和JDBC 两种外链Oracle数据库的方式均不支持；

eg.
JDBC:
`select count(1) from link_jdbc_wms_orders where orderdate = str_to_date('2022-12-13 00:00:00', '%Y-%m-%d %H:%i:%s') ;`
错误提示：
[HY000][1105] errCode = 2, detailMessage = UdfRuntimeException: JDBC executor sql has error: CAUSED BY: SQLDataException: ORA-01861: literal does not match format string  CAUSED BY: OracleDatabaseException: ORA-01861: literal does not match format string

`select count(1) from link_jdbc_wms_orders where orderdate = localtime() ;`
无报错，但该查询条件完全无效，（与不加条件一样）

`select count(1) from link_jdbc_wms_orders where orderdate > localtime() ;`
无报错，但该查询条件完全无效，（与不加条件一样）

ps: localtime() 换成 localtimestamp() 效果一样

ODBC：
`select count(1) from link_wms_orders where orderdate > str_to_date('2022-12-13 12:34:56', '%Y-%m-%d %H:%i:%s')`
错误提示：
[HY000][1105] errCode = 2, detailMessage = exec direct Error: HY000 [Oracle][ODBC][Ora]ORA-01861: literal does not match format string (1861)

`select count(1) from link_wms_orders where orderdate = localtime() ;`
无报错，但该查询条件完全无效，（与不加条件一样）

`select count(1) from link_wms_orders where orderdate > localtime() ;`
无报错，但该查询条件完全无效，（与不加条件一样）

附带JDBC建表语句：
`create EXTERNAL table if not exists link_jdbc_wms_orders
(
    serialkey         int comment '自增id',
    whseid            varchar(30) comment '仓库id',
    orderkey          varchar(10) comment '订单号',
    orderdate         datetime comment '下单日期',
    sortationlocation varchar(18) comment '电商平台名称'
) engine = jdbc comment "出库"
PROPERTIES (
"resource" = "jdbc_wms_oracle",
"table" = "orders",
"table_type"="oracle"
);`

附带ODBC建表语句：
`create EXTERNAL table if not exists link_wms_orders
(
    serialkey         int comment '自增id',
    whseid            varchar(30) comment '仓库id',
    orderkey          varchar(10) comment '订单号',
    orderdate         datetime comment '下单日期',
    sortationlocation varchar(18) comment '电商平台名称'
) engine = odbc comment "出库"
PROPERTIES (
    "host" =    "*",
    "port" =    "1521",
    "user" =    "*",
    "password" =    "*",
    "database" =    "scprd",
    "table" =    "WMWHSE1.orders",
    "driver" =     "oracle 19 odbc driver",
    "odbc_type" =    "oracle"
);`

**Comment by stevqin** [comment]
该问题可在 doris 1.1.2  doris 1.1.3   doris 1.1.4   doris 1.2.0 这几个版本中复现。

一路升级过来，问题依旧；

另外，1.1.2 的版本没有出现过BE蹦掉的现象，从升级 1.1.3 开始，BE经常会崩掉

**Comment by zhangstar333** [comment]
we will fix it as soon as possible
